src/Database/Connection.php

```diff
@@ -727,6 +728,39 @@ protected function _newLogger(StatementInterface $statement)
     }
 
     /**
+     * Checks if the driver supports EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @return bool
+     */
+    public function canExplain($query)
+    {
+        $query = (string)$query;
+
+        return preg_match('/^\s*(SELECT|INSERT|UPDATE|DELETE)\s/i', $query)
+            && $this->_driver->explainSQL($query) !== false;
+    }
+
+    /**
+     * Executes an EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @param array $params list or associative array of params to be interpolated in $query as values
+     * @param array $types list or associative array of types to be used for casting values in query
+     * @return \Cake\Database\StatementInterface
+     */
+    public function explain($query, array $params = [], array $types = [])
```

Comment:
Adding explain() method to query object is not a bad idea. However it cannot be of help with adding Explain button to SQL log panel. Because DebugKit cannot retrieve any query objects from QueryLogger.

If I take all your advice, I would need to write the following code in DebugKit.
```php
<?php
// In sql_log_panel.ctp
$connection = ConnectionManager::get('default');

$driver = $connection->driver();
?>
<?php foreach ($logger->queries() as $query): ?>
   <?php
    try {
        $explain = $driver->explainSQL($query->originalQuery);
    } catch (Exception $e) {
        $explain = false;
    }
    ?>
    <!-- Show query, took, rows here -->
    <?php if ($explain !== false): ?>
        <!-- Show explain button here -->
    <?php endif ?>
}
```
```php
// In ToolbarController
$connection = ConnectionManager::get('default');
$sql = $connection->driver()->explainSQL($sql);
$stmt = $connection->execute($sql, $params, $types);
```
Am I understand correctly?

By the way, If we have a query de-compiler, it would be really great.